File: Math/7. Reverse Integer/Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x >= 0:
            s = (int)(str(x)[::-1])  
        else:
            s = -(int)(str(-x)[::-1])
        if s >= -(2**31) and s <= (2**31)-1:
            return s
        else:
            return 0

File: Math/7. Reverse Integer/Reverse Integer.cpp
//Solution 1
class Solution {
public:
	int reverse(int x){
		long long res = 0;
		while (x != 0){
			int t = res * 10 + x % 10;
			if (t / 10 != res)
				return 0;
			res = t;
			t /= 10;
		}
		return res;
	}
};

//Solution 2
class Solution {
public:
	int reverse(int x){
		std::string str = std::to_string(x);
		if (str[0] == '-')
			std::reverse(str.begin() + 1, str.end());
		else
			std::reverse(str.begin(), str.end());
		try{
			return stoi(str);
		}
		catch (out_of_range ex){
			return 0;
		}
	}
};

//Solution 3

class Solution {
public:
    int reverse(int x) {
       long   r=0;    
         while(x!=0){
           long rev=x%10;
            r=r*10+rev;
            x=x/10;  
        }
         return (r > INT_MAX || r < INT_MIN) ? 0 : r;
        }   
};
